pkg/serviceaccounts/oauthclient/oauthclientregistry.go

```diff
@@ -0,0 +1,109 @@
+package oauthclient
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+
+	scopeauthorizer "github.com/openshift/origin/pkg/authorization/authorizer/scope"
+	oauthapi "github.com/openshift/origin/pkg/oauth/api"
+	"github.com/openshift/origin/pkg/oauth/registry/oauthclient"
+)
+
+const (
+	OAuthClientSecretAnnotation            = "serviceaccounts.openshift.io/oauth-client-secret"
```

Comment:
Ok, that's fine.

On Wed, May 18, 2016 at 1:51 PM, David Eads <notifications@github.com>
wrote:

> In pkg/serviceaccounts/oauthclient/oauthclientregistry.go
> <https://github.com/openshift/origin/pull/8878#discussion_r63750175>:
>
> > +import (
> > +	"fmt"
> > +	"strconv"
> > +	"strings"
> > +
> > +	kapi "k8s.io/kubernetes/pkg/api"
> > +	kclient "k8s.io/kubernetes/pkg/client/unversioned"
> > +	"k8s.io/kubernetes/pkg/serviceaccount"
> > +
> > +	scopeauthorizer "github.com/openshift/origin/pkg/authorization/authorizer/scope"
> > +	oauthapi "github.com/openshift/origin/pkg/oauth/api"
> > +	"github.com/openshift/origin/pkg/oauth/registry/oauthclient"
> > +)
> > +
> > +const (
> > +	OAuthClientSecretAnnotation            = "serviceaccounts.openshift.io/oauth-client-secret"
>
> Should this be oauth.openshift.io/client-secret? Is this oauth API or
> service account API?
>
> It only makes sense to specify this value on an SA token, so I saw it as
> SA owned thing that said, "use this for oauth".
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8878/files/1b4603924bf31e4db9b22f66721ace0c86ea02ee#r63750175>
>